### PR TITLE
HGCAL: restore distance cut in CLUE position calculation

### PR DESCRIPTION
**Pull request by amartelli**
This PR reintroduce the cut on the cells distance selected to compute the layer cluster position, that was probably lost in the migration from ImagingAlgo to CLUE.

For the record:
- the cut is 1.3cm for all silicon cells 
- the study was presented in the past in 
https://indico.cern.ch/event/776790/contributions/3230583/attachments/1761511/2865143/positionResolutionFor2DClusters_amartell_29Nov.pdf

This PR also addresses the open issue #31552

### CODE REVIEW

File: RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc [hunk 1]
```diff
@@ -214,6 +214,9 @@ math::XYZPoint HGCalCLUEAlgoT<T>::calculatePosition(const std::vector<int>& v, c
     float x_log = 0.f;
     float y_log = 0.f;
     for (auto i : v) {
+      //for silicon only just use 1+6 cells = 1.3cm for all thicknesses
+      if (distance(i, maxEnergyIndex, layerId, false) > positionDeltaRho_)
```
Comment (rovere) [inline RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc:218]: Maybe we could use the `distance2` to avoid `sqrt`?

Comment (amartelli) [reply RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc]: done, thanks

File: RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.h [hunk 2]
```diff
@@ -126,6 +128,7 @@ class HGCalCLUEAlgoT : public HGCalClusteringAlgoBase {
 private:
   // To compute the cluster position
   std::vector<double> thresholdW0_;
+  double positionDeltaRho2_;
```
Comment (felicepantaleo) [inline RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.h:131]: `const`?

**Review by felicepantaleo — CHANGES_REQUESTED** [review]

**Comment by felicepantaleo** [comment]
@cmsbuild please test

File: RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.h (hunk 2 above)
Comment (amartelli) [reply RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.h]: fixed, thanks

**Comment by slava77** [comment]
> Reco comparison results: 620 differences found in the comparisons

this corresponds to just 23234 and 28234

| WF # | Differences |
| --- | --- |
| all_OldVSNew_TTbar14TeV2026D49wf23234p0 | 295 |
| all_OldVSNew_TTbar14TeV2026D60wf28234p0 | 164 |

I'm curious why there are no differences in somewhat equivalent 23434.999, which has the same sim, but also has PU (50).
are the changes expected to be rather rare, or should we be looking for subtleties of the phase-2 premixing?

**Comment by amartelli** [comment]
Hi @slava77 
i don't find the folder with comparison plots for 23434.999, where do you see that there are no differences? 
(If I dig in the /files folder and retrieve the DQM or CMSSW files for 23234 and 23434.999 I see few differences between the two)

Differences in general should not be huge but not rare either! And should be there also with PU.
And I cannot see how premixing will hid changes in the layer clusters position...

**Comment by slava77** [comment]
>  i don't find the folder with comparison plots for 23434.999, where do you see that there are no differences?

https://tinyurl.com/y4b5rd9j from the bin-to-bin https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_2_X_2020-11-04-1100+951188/39716/dqm-histo-comparison-summary.html has nothing other than TimerService

while 23234 https://tinyurl.com/y5ylunnh has plenty of differences in physics quantities

**Comment by amartelli** [comment]
Thanks for the further info.

As I wrote, differences are expected, but not huge and quite often. 
I don't know if for comparison with PU there is some threshold to assume differences compatible with the statistical fluctuation (probably the mixing changes a bit between the two? or something else?) . If this is the case, maybe the differences from the PR are below threshold. 
Does this sound reasonable?

Since there are reasonable changes for the samples at 0PU, I assume there are also with PU, but indeed since they do not show up, maybe they are there and simply unspotted...

**Comment by slava77** [comment]
>  I don't know if for comparison with PU there is some threshold to assume differences compatible with the statistical fluctuation (probably the mixing changes a bit between the two? or something else?) . If this is the case, maybe the differences from the PR are below threshold.

there are no thresholds, any change in the histograms should be picked up

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/32022	994bb9f 
- code changes are in line with the PR description and the follow discussion 
- jenkins tests pass and comparisons with the baseline show differences only in phase-2 workflows (ignoring for now the lack of differences in 23434.999): differences start from `hgcalMulticlusters` (small increase in count by 0.4%) and propagate downstream to TICL tracksters and candidates (3% decrease in count)

**Comment by kpedro88** [comment]
+upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**